bokeh/embed.py

```diff
@@ -253,6 +253,10 @@ def notebook_div(model, notebook_comms_target=None):
     with _ModelInDocument(model):
         (docs_json, render_items) = _standalone_docs_json_and_render_items([model])
 
+    # Set the root_ids to only be the one we want to show in this div
+    doc_key = list(docs_json.keys())[0]
+    docs_json[doc_key]['roots']['root_ids'] = [model.ref['id']]
```

Comment:
I didn't dig in enough to understand the important questions here yet, but as a tactical matter, I'd suggest doing this transformation inside of `_standalone_docs_json_and_render_items()` or even inside of `Document.to_json`, as needed. (You would have to give `_standalone_docs_json_and_render_items()` and/or `Document.to_json` some kind of flag, perhaps, if you don't always want to do this transformation.)

The reason is that the above two lines are likely to become broken if anyone ever changes anything about the json. I'm not confident they are correct now; the ordering of `docs_json.keys()` isn't guaranteed as far as I know? It might be better if json-munging happened in the context of code that was assembling the json in the first place, so future-you can see all json-affecting code at once, instead of changing the json generating code and then discovering that the above two lines broke.

Anyway, aside from that ... I'm suspicious of this code even before I understand what bug we're fixing ;-) here are the two things that seem suspicious to me:
 * it appears that "which model to render" is already passed around as the model sent to `_standalone_docs_json_and_render_items()` and then below it's passed to `_div_for_render_item`, if everyone is properly looking at that parameter, why does `root_ids` matter? In effect we're creating a second copy of the same information maybe.
 * this creates a second "flavor" of serialized document, which is "the document, but with some missing root ids", and this special some-stuff-left-out flavor of serialized document will then have some sort of special side effect; vs. a situation where a serialized document is a serialized document is a serialized document. I'm imagining over on the JSON side there would be some big comment explaining how "in such-and-such scenario, the serialized document isn't really the entire document, so that's why we can use root_ids here even though there may be roots we don't care about..." (I don't know). I would think "which root ids to use in such-and-such special context" should be in another place, rather than part of the document serialization.

anyway, I dunno. let me go try to understand the bug ... gut reaction there has to be a better fix, though.